halyard-config/src/main/java/com/netflix/spinnaker/halyard/model/v1/Updateable.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2016 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.halyard.model.v1;
+
+import com.netflix.spinnaker.halyard.validate.v1.ValidateField;
+import com.netflix.spinnaker.halyard.validate.v1.Validator;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Field;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public abstract class Updateable<T> {
```

Comment:
`Updateable` implies a behavior or capability, much the same way `Cloneable` is below, not necessarily a type abstraction (like `KubernetesAccount` is-a `Account`). I suggest making this an interface with `update` as a Java default method, so that you have the much nicer looking `Account implements Updateable, Cloneable`.